components/speedreader/speedreader_rewriter_service.cc

```diff
@@ -90,9 +146,23 @@ void SpeedreaderRewriterService::OnStylesheetReady(const base::FilePath& path) {
 }
 
 bool SpeedreaderRewriterService::IsWhitelisted(const GURL& url) {
-  return backend_ == RewriterType::RewriterStreaming
-             ? speedreader_->IsReadableURL(url.spec())
-             : true;
+  if (backend_ == RewriterType::RewriterStreaming) {
+    return speedreader_->IsReadableURL(url.spec());
+  } else {
+    // Only HTTP is readable.
+    if (!url.SchemeIsHTTPOrHTTPS())
+      return false;
+
+    // @pes research has shown basically no landing pages are readable.
+    if (!url.has_path() || url.path_piece() == "/")
```

Comment:
just curious is `url.path_piece()` really needed?

Reply:
didn't need to be. changed to `url.path()`